test/common/filter/tcp_proxy_test.cc

```diff
@@ -741,6 +739,43 @@ TEST_F(TcpProxyTest, UpstreamConnectionLimit) {
   EXPECT_EQ(access_log_data_, "UO");
 }
 
+// Tests that the idle timer closes both connections, and gets updated when either
+// connection has activity
+TEST_F(TcpProxyTest, IdleTimeout) {
+  envoy::api::v2::filter::network::TcpProxy config = defaultConfig();
+  config.mutable_idle_timeout()->set_seconds(1);
+  setup(1, config);
+
+  Network::BytesSentCb downstream_bytes_sent_cb;
+  Network::BytesSentCb upstream_bytes_sent_cb;
+  EXPECT_CALL(filter_callbacks_.connection_, addBytesSentCallback(_))
```

Comment:
You might consider adding this capability to the base mock, and then adding a `raiseBytesSentCallbacks` method on the base mock (like we do for other events). I'm guessing someone else is going to want to use this later.